File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/88_merge-sorted-array/solutions/java/solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int p1 = m - 1; // Pointer for nums1
        int p2 = n - 1; // Pointer for nums2
        int i = m + n - 1; // Pointer for the end of nums1 array

        // Merge in reverse order
        while (p1 >= 0 && p2 >= 0) {
            if (nums1[p1] > nums2[p2]) {
                nums1[i--] = nums1[p1--];
            } else {
                nums1[i--] = nums2[p2--];
            }
        }

        // Add remaining elements from nums2 if there are any
        while (p2 >= 0) {
            nums1[i--] = nums2[p2--];
        }
    }
}

File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/88_merge-sorted-array/solutions/c/solution.c
#include <stdio.h>

void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
    int i = m - 1; // Last valid element of nums1
    int j = n - 1; // Last element of nums2
    int k = m + n - 1; // End of merged array

    while (j >= 0) {
        if (i >= 0 && nums1[i] >= nums2[j]) {
            nums1[k] = nums1[i];
            i--;
        } else {
            nums1[k] = nums2[j];
            j--;
        }
        k--;
    }
}
